test/jtreg-ext/requires/VMProps.java

```diff
@@ -499,15 +508,67 @@ protected String dockerSupport() {
            }
          }
 
+        log("dockerSupport(): returning isSupported = " + isSupported);
         return "" + isSupported;
     }
 
+    // Returns comma-separated file names for stdout and stderr.
+    private String redirectOutputToLogFile(String msg, ProcessBuilder pb, String fileNameBase) {
+        if (!Boolean.getBoolean("jtreg.log.vmprops")) {
```

Comment:
Why not dump output always? It is printed only if exitcode is not zero anyway and doesn't add noise.
However not having it in the case of failure require reproducing the bug.

Reply:
We have to be consistent. Since we agreed in earlier discussion to use property jtreg.log.vmprops as a guard for logging we should also use it for writing into a file. The user who runs the tests can always specify this property in the test execution script.

Reply:
If we take the approach of always logging to vmprops.log then this makes sense. If no objections I will always redirect output into a file.